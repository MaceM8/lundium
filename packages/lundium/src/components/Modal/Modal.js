import React from 'react';
import { cx } from 'ramda-extension';
import PropTypes from 'prop-types';

import Icon from '../Icon/Icon';
import Button from '../Button';
import PortalWithBackdrop from '../PortalWithBackdrop';

const stopPropagation = e => {
	// NOTE: Click event would be propagated up to backdrop
	e.stopPropagation();
};
const Modal = ({
	children,
	isVisible,
	className,
	titleCenter,
	title,
	onCancel,
	// eslint-disable-next-line no-unused-vars
	modalId,
}) => (
	<PortalWithBackdrop isVisible={isVisible} onBackdropClick={onCancel}>
		<div className={cx('modal', className)} onClick={stopPropagation}>
			<Button kind="blank" className="modal__close" onClick={onCancel}>
				<Icon type="cancel" />
			</Button>
			<div className={cx('modal__title', { ['text-center']: titleCenter })}>{title}</div>
			<div className="modal__content">{children}</div>
		</div>
	</PortalWithBackdrop>
);

Modal.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.any.isRequired,
	/** Class name for modal root */
	className: PropTypes.node,
	/** Is modal visible */
	isVisible: PropTypes.bool,
	/** Modal id */
	modalId: PropTypes.string,
	/** On cancel click */
	onCancel: PropTypes.func,
	/** Modals title */
	title: PropTypes.node.isRequired,
	/** Is title in center */
	titleCenter: PropTypes.bool,
};

// TODO: add transition
export default Modal;
