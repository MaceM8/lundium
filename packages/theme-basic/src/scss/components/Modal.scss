@import '../../scss/core';

/*
 -  Modal
 -  (<div class="modal" />)
 */


.modal {
	position: relative;
	width: 100%;
	min-height: 100px;
	padding: logical ( $spacer * 2 ) $spacer ( $spacer * 3 );
	border-radius: $border-radius;
	background-color: $modal-bg;
	box-shadow: $modal-box-shadow;
	transform: translateY(-40px);
	transition: transform ( $transition-speed * 2 ) $transition-easing;

	// Media queries
	@include media-breakpoint-up(sm) {
		max-width: $modal-width;
		padding: logical ( $spacer * 2 ) $spacer ( $spacer * 3 );
	}

	&__close {
		position: absolute;
		inset-block-start: $spacer;
		inset-inline-end: $spacer;
		min-width: auto;
		width: 30px;
		height: 30px;
		padding: logical 0;

		.icon--type-cancel {
			padding-inline-end: 0;
		}
	}

	&__title {
		font-weight: $modal-title-font-weight;
		font-size: $modal-title-font-size;
		margin-block-end: $spacer;
	}
}