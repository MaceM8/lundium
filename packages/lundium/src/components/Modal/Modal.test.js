import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

describe('<Modal />', () => {
	const requiredProps = { title: 'Modal title', children: <div>Some content</div> };

	it('shallow', () => {
		const wrapper = shallow(<Modal {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
