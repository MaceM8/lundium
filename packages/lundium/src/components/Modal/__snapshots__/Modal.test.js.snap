// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> shallow 1`] = `
<PortalWithBackdrop>
  <div
    className="modal"
    onClick={[Function]}
  >
    <ForwardRef
      className="modal__close"
      kind="blank"
    >
      <Icon
        type="cancel"
      />
    </ForwardRef>
    <div
      className="modal__title"
    >
      Modal title
    </div>
    <div
      className="modal__content"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</PortalWithBackdrop>
`;
